includes/object-cache.php

```diff
@@ -2138,6 +2242,50 @@ protected function handle_exception( $exception ) {
     public function __get( $name ) {
         return isset( $this->{$name} ) ? $this->{$name} : null;
     }
+
+    /**
+     * Calculates how much time has elapsed since $start_time.
+     *
+     * @param float $start_time
+     * @return float
+     */
+    private function elapsed( $start_time ) {
+        return microtime(true) - $start_time;
+    }
+
+    /**
+     * Invoke the `redis_object_cache_trace` hook.
+     *
+     * @param  string             $command
+     * @param  string             $group
+     * @param  array[string]array $keyValues
+     * @param  float              $duration
+     * @return void
+     */
+    private function trace_command ( $command, $group, $keyValues, $duration ) {
+        if ( !function_exists( 'do_action' ) ) {
+            return;
+        }
```

Comment:
In addition to a `do_action()` check, we should probably also call `has_action('redis_object_cache_trace')`.

https://developer.wordpress.org/reference/functions/has_action/

Reply:
Sidenote: checking `has_action` will avoid firing the `all` hook.  Performance would be gained by checking this first but it might be unexpected.